### ISSUE TITLE
Cross-references for images

### ISSUE THREAD

**Issue by AndreWin** [asciidoctor/asciidoctor#2472]
Hello!

```
= Моя статья
:xrefstyle: short
:imagesdir: ./imgs

Некоторый текст. Смотри рис. <<img_01>>.

.Какая-то красивая схема
[#img_01]
image::01.png[, 80%,align="center"]
```

![image](https://user-images.githubusercontent.com/7497378/32506286-319b2a02-c3f5-11e7-8477-10544d6d2d55.png)

I wll get` Некоторый текст. Смотри рис. Figure 1`. I'd like to make `Некоторый текст. Смотри рис. 1`.

How can I make cross-reference to image without `Figure`? Or change `Figure` on my own text on Russian?

Best regards,
Andrey.

**Comment by DavidGamba** [comment]
Take a look at: http://asciidoctor.org/docs/user-manual/#customizing-labels

`:figure-caption: Рис`

**Comment by AndreWin** [comment]
@DavidGamba, thanks a lot!

How can I make:
1. `Рис. 1. Красивая схема` -- under image, starts with capital letter,
2. `см. <<img_01>>` &rarr; `см. рис. 1` -- this `рис.` starts with small letter.

Is this possible?

**Comment by DavidGamba** [comment]
http://asciidoctor.org/docs/user-manual/#customizing-the-cross-reference-text

Seems like you could possibly do `:figure-refsig: рис.` but I haven't tested it myself.

**Comment by AndreWin** [comment]
@DavidGamba, this didn't help me. It seems to me `:figure-refsig:` just doesn't work.

**Comment by DavidGamba** [comment]
`:figure-refsig:` is not implemented, it was just a guess, sorry about that. See: https://github.com/asciidoctor/asciidoctor/issues/2212 for details into how it will actually be implemented into the future.

At the moment I don't think that what you want is possible.

**Comment by mojavelinux** [comment]
You're correct, the refsig attributes are not yet implemented for non-section blocks. The reason is, we're testing the refsig idea on sections first. If that works out, as it seems it is, then we'll continue and implement this for all references.

Right now, figure-caption is used for both the reference and the block title. You can see the logic here: https://github.com/asciidoctor/asciidoctor/blob/911d0bd509f369e9da15d2bb71f81aecb7c45fec/lib/asciidoctor/abstract_block.rb#L335-L376

It's always possible to customize the HTML by supplying a custom template. In this case, you'd want to supply a custom template for the inline_anchor node. As of 1.5.6, you have access to the node object of the target, so you can check if it is an image and output text the way you want. That's the really important part of the change that came in 1.5.6.

See https://github.com/asciidoctor/asciidoctor.org/issues/80 to learn about custom templates.

**Comment by SergeiKuznetsov** [comment]
It works
```
= Моя статья
:xrefstyle: short
:imagesdir: ./imgs

Некоторый текст. Смотри <<img_01>>.

[caption="Рисунок {counter:figure}. ", reftext="рис. {figure}"]
.Какая-то красивая схема
[#img_01]
image::01.png[, 80%,align="center"]
```

**Comment by SergeiKuznetsov** [comment]
A custom counter has to be added to each figure. Only in this way.

**Comment by fiddlededee** [comment]
Threre is another way, that allows to tailor each reference. For example, reference at the beginning of the sentence should still start with upper case letter.

```
:xrefstyle: short

We can put it this way: 

Smth is shown on the {set:figure-caption:figure} <<img_01>>.

Or this way:

{set:figure-caption:Figure} <<img_01>> shows smth.

:figure-caption: Figure
.Example image
[#img_01]
image::01.png[, 80%,align="center"]
```

In the previous example we can get smth case aware: Что-то изображено на рисунке 1. Abbreviating the word `Figure` in my opinion makes text less beautiful.

Although it would be great to see smth like:

* figure-caption attribute (exists)
* ability to override figure-caption (exists)
* figure-refsig attribute (doesn't exist)
* ability to override figure-refsig in referenece like <<img_01, figure-refsig=...>> (doesn't exist)

**Comment by fiddlededee** [comment]
You may use smth like

```
:figure-caption: _figure_caption.adoc

include::{figure-caption}[]
```

And `_table_caption.adoc` may contain the above magic string.

At least you can easily change it and remove if `figure-refsig` will appear.
